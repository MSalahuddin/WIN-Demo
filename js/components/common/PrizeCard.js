import React from 'react';
import styled from 'styled-components/native';
import PropTypes from 'prop-types';
import {Platform} from 'react-native';
import Text, { SIZE } from './Text';
import { scale, scaleHeight, getWindowWidth } from '../../platformUtils';
import { color } from '../../styles';
import { ticket } from '../../../assets/images';
import { convertNumberToStringWithComma } from '../../utils';

const cardWidth = (getWindowWidth() - 3 * scale(16)) / 2;
const Container = styled.TouchableOpacity`
  background-color: ${color.white};
  border-radius: ${scale(8)};
  margin-top: ${scale(12)};
  margin-horizontal: ${scale(5)};
  shadow-color: ${color.blackShadow};
  shadow-offset: 4px 4px;
  shadow-opacity: ${1};
  shadow-radius: 2px;
  width: ${cardWidth};
  overflow: hidden;
  flex-direction: column;
`;

const PrizeImage = styled.Image`
  width: ${cardWidth};
  height: ${cardWidth};
  overflow: hidden;
`;

const TitleBar = styled.View`
  flex: 1;
  padding-horizontal: ${scale(20)};
  padding-vertical: ${scaleHeight(10)};
  justify-content: center;
  align-items: center;
`;

const TicketBar = styled.View`
  flex: 1;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding-horizontal: ${scale(20)};
  padding-vertical: ${scaleHeight(10)};
  background-color: ${color.grayWhite};
`;

const TicketWrapper = styled.View`
  flex: 1;
  flex-direction: row;
  align-items: center;
  justify-content: center;
`;

const TicketImage = styled.Image`
  height: ${scaleHeight(14)};
  width: ${scale(18)};
  margin-right: ${scale(5)};
`;

const TicketText = styled(Text)`
  padding-top: ${Platform.OS==='android'?scaleHeight(0):scaleHeight(4)};
`;

const PrizeCard = ({ title, imageUrl, ticketQuantity, onPress }) => {
  return (
    <Container testID="prize-card-button" activeOpacity={1} onPress={onPress}>
      <PrizeImage source={{ uri: imageUrl }} resizeMode="cover" />
      <TitleBar>
        <Text size={SIZE.XXSMALL} alignCenter ellipsizeMode="tail">
          {title}
        </Text>
      </TitleBar>
      <TicketBar>
        <TicketWrapper>
          <TicketImage source={ticket} />
          <TicketText size={SIZE.XXSMALL} ellipsizeMode="tail">
            {`x${convertNumberToStringWithComma(ticketQuantity)}`}
          </TicketText>
        </TicketWrapper>
      </TicketBar>
    </Container>
  );
};

PrizeCard.defaultProps = {
  ticketQuantity: 0,
  onPress: () => {}
};

PrizeCard.propTypes = {
  title: PropTypes.string.isRequired,
  imageUrl: PropTypes.string.isRequired,
  ticketQuantity: PropTypes.number,
  onPress: PropTypes.func
};

export default PrizeCard;
