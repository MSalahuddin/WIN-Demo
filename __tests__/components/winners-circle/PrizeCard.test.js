import React from 'react';
import { render, fireEvent } from 'react-native-testing-library';

import PrizeCard from '../../../js/components/winners-circle/PrizeCard';
import MockProvider from '../../MockProvider';

describe('Component PrizeCard', () => {
  const onPress = jest.fn();
  const onPressInfo = jest.fn();

  test('renders regular PrizeCard correctly', () => {
    const output = render(
      <MockProvider>
        <PrizeCard prizeName="" description="" value={0} imageUrl="" onPress={onPress} onPressInfo={onPressInfo} />
      </MockProvider>
    );
    expect(output).toMatchSnapshot();
  });

  test('renders popular PrizeCard on sale', () => {
    const output = render(
      <MockProvider>
        <PrizeCard
          prizeName=""
          description=""
          value={10}
          imageUrl=""
          isOnSale
          saleTicketValue={0}
          onPressInfo={onPressInfo}
          onPress={onPress}
        />
      </MockProvider>
    );
    expect(output).toMatchSnapshot();
  });

  test('calls onPressInfo when info button is pressed', () => {
    const { getByTestId } = render(
      <MockProvider>
        <PrizeCard
          prizeName="a"
          description=""
          value={10}
          imageUrl=""
          isOnSale
          saleTicketValue={0}
          onPressInfo={onPressInfo}
          onPress={onPress}
        />
      </MockProvider>
    );
    fireEvent(getByTestId('winner-info-button'), 'press');
    expect(onPressInfo).toHaveBeenCalled();
  });
});
