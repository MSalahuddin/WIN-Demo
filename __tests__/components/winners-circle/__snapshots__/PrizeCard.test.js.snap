// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component PrizeCard renders popular PrizeCard on sale 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 12,
        "elevation": 8,
        "marginTop": 18,
        "paddingBottom": 12,
        "shadowColor": "rgba(53, 53, 53, 0.3)",
        "shadowOffset": Object {
          "height": 4,
          "width": 4,
        },
        "shadowOpacity": 1,
        "shadowRadius": 2,
        "width": 339,
      },
    ]
  }
>
  <Image
    resizeMode="cover"
    source={
      Object {
        "uri": "",
      }
    }
    style={
      Array [
        Object {
          "borderTopLeftRadius": 12,
          "borderTopRightRadius": 12,
          "height": 278.432,
          "width": 339,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "marginBottom": 20,
          "marginTop": 36,
          "paddingHorizontal": 24,
        },
      ]
    }
  >
    <Text
      alignCenter={false}
      allowFontScaling={true}
      color="#222222"
      fontFamily="SEMI BOLD"
      isUnderlined={false}
      isUppercase={false}
      maxFontSizeMultiplier={1.3}
      numberOfLines={1}
      size="XXSMALL"
      style={
        Array [
          Object {
            "color": "#222222",
            "fontFamily": "Calibre-Semibold",
            "fontSize": 21,
            "justifyContent": "center",
            "textAlign": "left",
            "textDecorationColor": "#222222",
            "textDecorationLine": "none",
            "textDecorationStyle": "solid",
            "textTransform": "none",
          },
          Object {
            "marginTop": 0,
          },
        ]
      }
    >
      
    </Text>
    <Text
      alignCenter={false}
      allowFontScaling={true}
      color="#8B8B8B"
      fontFamily="REGULAR"
      isUnderlined={false}
      isUppercase={false}
      marginTop={6}
      maxFontSizeMultiplier={1.3}
      numberOfLines={1}
      size="XXSMALL"
      style={
        Array [
          Object {
            "color": "#8B8B8B",
            "fontFamily": "CalibreRegular",
            "fontSize": 21,
            "justifyContent": "center",
            "textAlign": "left",
            "textDecorationColor": "#8B8B8B",
            "textDecorationLine": "none",
            "textDecorationStyle": "solid",
            "textTransform": "none",
          },
          Object {
            "marginTop": 12,
          },
        ]
      }
    >
      
    </Text>
    <Text
      alignCenter={false}
      allowFontScaling={true}
      color="#353535"
      fontFamily="MEDIUM"
      isUnderlined={false}
      isUppercase={false}
      marginTop={8}
      maxFontSizeMultiplier={1.3}
      numberOfLines={1}
      size="XXSMALL"
      style={
        Array [
          Object {
            "color": "#353535",
            "fontFamily": "Calibre-Medium",
            "fontSize": 21,
            "justifyContent": "center",
            "textAlign": "left",
            "textDecorationColor": "#353535",
            "textDecorationLine": "none",
            "textDecorationStyle": "solid",
            "textTransform": "none",
          },
          Object {
            "marginTop": 16,
          },
        ]
      }
    >
      Redeem for
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    hitSlop={
      Object {
        "bottom": 12,
        "left": 12,
        "right": 12,
        "top": 12,
      }
    }
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "marginLeft": 24,
        "marginTop": 16,
        "opacity": 1,
        "position": "absolute",
        "right": 12,
      }
    }
    testID="winner-info-button"
  >
    <Image
      opacity={1}
      resizeMode="contain"
      size={24}
      source={1}
      style={
        Array [
          Object {
            "height": 36,
            "width": 36,
          },
        ]
      }
    />
  </View>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginHorizontal": 10,
          "marginVertical": 10,
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(160,160,160,1)",
          "borderRadius": 106,
          "height": 96,
          "justifyContent": "center",
          "opacity": 1,
          "width": 278,
        }
      }
      testID="redeem-button"
    >
      <View
        borderRadius={53}
        colorScheme="rgba(209,209,209,1)"
        height={48}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(209,209,209,1)",
              "borderRadius": 106,
              "height": 86,
              "justifyContent": "center",
              "width": 272,
            },
          ]
        }
        width={139}
      >
        <View
          borderRadius={53}
          colorScheme="rgba(183,183,183,1)"
          height={48}
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(183,183,183,1)",
                "borderRadius": 126,
                "height": 82,
                "justifyContent": "center",
                "marginTop": 8,
                "width": 272,
              },
            ]
          }
          width={139}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
              ]
            }
          >
            <Image
              resizeMode="contain"
              size={29}
              source={1}
              style={
                Array [
                  Object {
                    "height": 43.5,
                    "marginRight": 16,
                    "width": 43.5,
                  },
                ]
              }
            />
            <View>
              <Text
                alignCenter={false}
                allowFontScaling={true}
                color="#E5E5E5"
                fontFamily="BOLD"
                isUnderlined={false}
                isUppercase={false}
                marginTop={8}
                maxFontSizeMultiplier={1.3}
                size="SMALL"
                style={
                  Array [
                    Object {
                      "color": "#E5E5E5",
                      "fontFamily": "CalibreBold",
                      "fontSize": 27,
                      "justifyContent": "center",
                      "textAlign": "left",
                      "textDecorationColor": "#E5E5E5",
                      "textDecorationLine": "none",
                      "textDecorationStyle": "solid",
                      "textTransform": "none",
                    },
                    Object {
                      "marginTop": 16,
                    },
                  ]
                }
              >
                10
              </Text>
              0
            </View>
          </View>
          0
        </View>
      </View>
    </View>
  </View>
  <Image
    resizeMode="contain"
    source={1}
    style={
      Array [
        Object {
          "height": 116,
          "left": -4,
          "position": "absolute",
          "top": -4,
          "width": 120,
        },
      ]
    }
  />
</View>
`;

exports[`Component PrizeCard renders regular PrizeCard correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 12,
        "elevation": 8,
        "marginTop": 18,
        "paddingBottom": 12,
        "shadowColor": "rgba(53, 53, 53, 0.3)",
        "shadowOffset": Object {
          "height": 4,
          "width": 4,
        },
        "shadowOpacity": 1,
        "shadowRadius": 2,
        "width": 339,
      },
    ]
  }
>
  <Image
    resizeMode="cover"
    source={
      Object {
        "uri": "",
      }
    }
    style={
      Array [
        Object {
          "borderTopLeftRadius": 12,
          "borderTopRightRadius": 12,
          "height": 278.432,
          "width": 339,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "marginBottom": 20,
          "marginTop": 36,
          "paddingHorizontal": 24,
        },
      ]
    }
  >
    <Text
      alignCenter={false}
      allowFontScaling={true}
      color="#222222"
      fontFamily="SEMI BOLD"
      isUnderlined={false}
      isUppercase={false}
      maxFontSizeMultiplier={1.3}
      numberOfLines={1}
      size="XXSMALL"
      style={
        Array [
          Object {
            "color": "#222222",
            "fontFamily": "Calibre-Semibold",
            "fontSize": 21,
            "justifyContent": "center",
            "textAlign": "left",
            "textDecorationColor": "#222222",
            "textDecorationLine": "none",
            "textDecorationStyle": "solid",
            "textTransform": "none",
          },
          Object {
            "marginTop": 0,
          },
        ]
      }
    >
      
    </Text>
    <Text
      alignCenter={false}
      allowFontScaling={true}
      color="#8B8B8B"
      fontFamily="REGULAR"
      isUnderlined={false}
      isUppercase={false}
      marginTop={6}
      maxFontSizeMultiplier={1.3}
      numberOfLines={1}
      size="XXSMALL"
      style={
        Array [
          Object {
            "color": "#8B8B8B",
            "fontFamily": "CalibreRegular",
            "fontSize": 21,
            "justifyContent": "center",
            "textAlign": "left",
            "textDecorationColor": "#8B8B8B",
            "textDecorationLine": "none",
            "textDecorationStyle": "solid",
            "textTransform": "none",
          },
          Object {
            "marginTop": 12,
          },
        ]
      }
    >
      
    </Text>
    <Text
      alignCenter={false}
      allowFontScaling={true}
      color="#353535"
      fontFamily="MEDIUM"
      isUnderlined={false}
      isUppercase={false}
      marginTop={8}
      maxFontSizeMultiplier={1.3}
      numberOfLines={1}
      size="XXSMALL"
      style={
        Array [
          Object {
            "color": "#353535",
            "fontFamily": "Calibre-Medium",
            "fontSize": 21,
            "justifyContent": "center",
            "textAlign": "left",
            "textDecorationColor": "#353535",
            "textDecorationLine": "none",
            "textDecorationStyle": "solid",
            "textTransform": "none",
          },
          Object {
            "marginTop": 16,
          },
        ]
      }
    >
      Redeem for
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    hitSlop={
      Object {
        "bottom": 12,
        "left": 12,
        "right": 12,
        "top": 12,
      }
    }
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "marginLeft": 24,
        "marginTop": 16,
        "opacity": 1,
        "position": "absolute",
        "right": 12,
      }
    }
    testID="winner-info-button"
  >
    <Image
      opacity={1}
      resizeMode="contain"
      size={24}
      source={1}
      style={
        Array [
          Object {
            "height": 36,
            "width": 36,
          },
        ]
      }
    />
  </View>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginHorizontal": 10,
          "marginVertical": 10,
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(160,160,160,1)",
          "borderRadius": 106,
          "height": 96,
          "justifyContent": "center",
          "opacity": 1,
          "width": 278,
        }
      }
      testID="redeem-button"
    >
      <View
        borderRadius={53}
        colorScheme="rgba(209,209,209,1)"
        height={48}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(209,209,209,1)",
              "borderRadius": 106,
              "height": 86,
              "justifyContent": "center",
              "width": 272,
            },
          ]
        }
        width={139}
      >
        <View
          borderRadius={53}
          colorScheme="rgba(183,183,183,1)"
          height={48}
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(183,183,183,1)",
                "borderRadius": 126,
                "height": 82,
                "justifyContent": "center",
                "marginTop": 8,
                "width": 272,
              },
            ]
          }
          width={139}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
              ]
            }
          >
            <Image
              resizeMode="contain"
              size={29}
              source={1}
              style={
                Array [
                  Object {
                    "height": 43.5,
                    "marginRight": 16,
                    "width": 43.5,
                  },
                ]
              }
            />
            <View>
              <Text
                alignCenter={false}
                allowFontScaling={true}
                color="#E5E5E5"
                fontFamily="BOLD"
                isUnderlined={false}
                isUppercase={false}
                marginTop={8}
                maxFontSizeMultiplier={1.3}
                size="SMALL"
                style={
                  Array [
                    Object {
                      "color": "#E5E5E5",
                      "fontFamily": "CalibreBold",
                      "fontSize": 27,
                      "justifyContent": "center",
                      "textAlign": "left",
                      "textDecorationColor": "#E5E5E5",
                      "textDecorationLine": "none",
                      "textDecorationStyle": "solid",
                      "textTransform": "none",
                    },
                    Object {
                      "marginTop": 16,
                    },
                  ]
                }
              >
                0
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
